### PR TITLE
Use member initialization list instead of the constructor's body

### PR DESCRIPTION
**Pull request by lupoDharkael**
solves [this](https://github.com/minetest/minetest/issues/6368)

### CODE REVIEW

**Comment by SmallJoker** [comment]
Where's the profit?

@paramat might be interested in this PR.

**Comment by lupoDharkael** [comment]
I don't expect any big performance improvement, but:
1. It's more idiomatic and correct C++
2. Prevents not needed memory copy
3. It's an easy and non intrusive fix
4. Doesn't break compatibility
5. You don't loose anything by merging this

File: src/mapgen_carpathian.cpp [hunk 1]
```diff
@@ -106,24 +106,24 @@ MapgenCarpathian::~MapgenCarpathian()
...
+    np_hills_terrain (NoiseParams(1,  1,  v3f(1301, 1301, 1301), 1692,  5, 0.5,  2.0)),
+    np_ridge_terrain (NoiseParams(1,  1,  v3f(1889, 1889, 1889), 3568,  5, 0.5,  2.0)),
+    np_step_terrain  (NoiseParams(1,  1,  v3f(1889, 1889, 1889), 4157,  5, 0.5,  2.0)),
+    np_hills         (NoiseParams(0,  3,  v3f(257,  257,  257),  6604,  6, 0.5,  2.0)),
+    np_ridge_mnt     (NoiseParams(0,  12, v3f(743,  743,  743),  5520,  6, 0.7,  2.0)),
+    np_step_mnt      (NoiseParams(0,  8,  v3f(509,  509,  509),  2590,  6, 0.6,  2.0)),
+    np_mnt_var       (NoiseParams(0,  1,  v3f(499,  499,  499),  2490,  5, 0.55, 2.0)),
+    np_cave1         (NoiseParams(0,  12, v3f(61,   61,   61),   52534, 3, 0.5,  2.0)),
+    np_cave2         (NoiseParams(0,  12, v3f(67,   67,   67),   10325, 3, 0.5,  2.0)),
+    np_cavern        (NoiseParams(0,  1,  v3f(384,  128,  384),  723,   5, 0.63, 2.0))
```
Comment (SmallJoker) [inline src/mapgen_carpathian.cpp]: Lost proper tabulator indents.

File: src/mapgen_flat.cpp [hunk 2]
```diff
@@ -81,12 +81,12 @@ MapgenFlat::~MapgenFlat()
 }
 
 
-MapgenFlatParams::MapgenFlatParams()
+MapgenFlatParams::MapgenFlatParams():
+        np_terrain      (NoiseParams(0, 1,   v3f(600, 600, 600), 7244,  5, 0.6, 2.0)),
+        np_filler_depth (NoiseParams(0, 1.2, v3f(150, 150, 150), 261,   3, 0.7, 2.0)),
+        np_cave1        (NoiseParams(0, 12,  v3f(61,  61,  61),  52534, 3, 0.5, 2.0)),
+        np_cave2        (NoiseParams(0, 12,  v3f(67,  67,  67),  10325, 3, 0.5, 2.0))
```
Comment (SmallJoker) [inline src/mapgen_flat.cpp]: More broken indents.

File: src/mapgen_carpathian.cpp [hunk 3]
```diff
@@ -106,24 +106,24 @@ MapgenCarpathian::~MapgenCarpathian()
 }
 
 
-MapgenCarpathianParams::MapgenCarpathianParams()
+MapgenCarpathianParams::MapgenCarpathianParams():
+    np_base          (NoiseParams(12, 1,  v3f(2557, 2557, 2557), 6538,  4, 0.8,  0.5)),
```
Comment (nerzhul) [inline src/mapgen_carpathian.cpp]: if i remember the NoiseParams call is implicit and can be removed no ?

**Comment by nerzhul** [comment]
@SmallJoker impact is very tiny as mapgen is created once

**Comment by lupoDharkael** [comment]
Hope it's ok now.

**Comment by paramat** [comment]
Even if there's no performance gain it's still good habit and more correct coding, so :+1: if tested.

**Pull request merged**